tests/rados/pool_tests.py

```diff
@@ -1131,6 +1143,117 @@ def run(ceph_cluster, **kw):
                 rados_obj.log_cluster_health()
         return 0
 
+    if config.get("Verify_osd_in_out_behaviour"):
+        log.info(
+            "Test to verify the behaviour of the OSD when it is marked out and back in to the cluster"
+        )
+        try:
+            osd_tree = rados_obj.run_ceph_command(cmd="ceph osd tree")
+            osd_list = []
+            for entry in osd_tree["nodes"]:
+                if entry["id"] > 0:
+                    osd_list.append(entry["id"])
+            selected_osd = random.choice(osd_list)
+
+            log.info(
+                f"OSDs on the cluster : {osd_list}\n Selected OSD for testing : {selected_osd}"
+            )
+            osd_tree = rados_obj.run_ceph_command(cmd="ceph osd tree")
+            for entry in osd_tree["nodes"]:
+                if entry["id"] == selected_osd:
+                    log.info(f"Selected OSD details before marking out : {entry}")
+                    selected_osd_details_init = entry
+                    break
+
+            # Making the OSD out of the cluster
+            if not rados_obj.update_osd_state_on_cluster(
+                osd_id=selected_osd, state="out"
+            ):
+                log.error(f"OSD.{selected_osd} not marked out error")
+                raise Exception("OSD not marked out error")
+            log.info(f"OSD: {selected_osd} marked out successfully. ")
+
+            osd_tree = rados_obj.run_ceph_command(cmd="ceph osd tree")
+            for entry in osd_tree["nodes"]:
+                if entry["id"] == selected_osd:
+                    log.info(f"Selected OSD details After marking out : {entry}")
+                    selected_osd_details_post = entry
+                    break
```

Comment:
Just a suggestion, we can skip this code block by utilizing exiting method for `ceph osd df` and use it to fetch the stat for the concerned osd itself
```
[root@ceph-hakumar-mm6odk-node7 ~]# ceph osd df osd.0
ID  CLASS  WEIGHT   REWEIGHT  SIZE    RAW USE  DATA    OMAP  META    AVAIL   %USE  VAR   PGS  STATUS
 0    hdd  0.02440   1.00000  25 GiB   46 MiB  10 MiB   0 B  36 MiB  25 GiB  0.18  1.00   10      up
                       TOTAL  25 GiB   46 MiB  10 MiB   0 B  36 MiB  25 GiB  0.18                   
MIN/MAX VAR: 1.00/1.00  STDDEV: 0
[root@ceph-hakumar-mm6odk-node7 ~]# ceph osd df osd.0 -f json-pretty

{
    "nodes": [
        {
            "id": 0,
            "device_class": "hdd",
            "name": "osd.0",
            "type": "osd",
            "type_id": 0,
            "crush_weight": 0.0243988037109375,
            "depth": 2,
            "pool_weights": {},
            "reweight": 1,
            "kb": 26210304,
            "kb_used": 46600,
            "kb_used_data": 10240,
            "kb_used_omap": 0,
            "kb_used_meta": 36352,
            "kb_avail": 26163704,
            "utilization": 0.17779267268323176,
            "var": 1,
            "pgs": 10,
            "status": "up"
        }
    ],
    "stray": [],
    "summary": {
        "total_kb": 26210304,
        "total_kb_used": 46600,
        "total_kb_used_data": 10240,
        "total_kb_used_omap": 0,
        "total_kb_used_meta": 36352,
        "total_kb_avail": 26163704,
        "average_utilization": 0.17779267268323176,
        "min_var": 1,
        "max_var": 1,
        "dev": 0
    }
}```

Reply:
I'll have a backlog task to create a new method for this use-case, something like `def get_osd_details` to fetch all the details for a particular OSD.

Reply:
Backlog task : https://issues.redhat.com/browse/RHCEPHQE-15034